### DISCUSSION TITLE
Create custom library

### DISCUSSION THREAD

**Discussion by marvinschmidtke** [nextgenhealthcare/connect#4488] [Q&A] [answered]
Hi guys,

im trying since yesterday to get this simple Mirth tutorial working:
https://www.mirthcorp.com/community/wiki/display/mirth/How+to+create+and+invoke+custom+Java+code+in+Mirth+Connect

I've created the simple POJO file and converted it via eclipse and was also able to import it into another project:

![image](https://user-images.githubusercontent.com/19649604/114523821-f231ef00-9c44-11eb-8ead-40f18b4fb725.png)

When im trying to use this Class now in Mirth i get the following error:
![image](https://user-images.githubusercontent.com/19649604/114523936-142b7180-9c45-11eb-8b8b-cf2537c758c9.png)

Can someone please tell me what is wrong with my jar file? There shouldn't be many pitfalls in this simple tutorial ?

I've made sure that the library is imported:
![image](https://user-images.githubusercontent.com/19649604/114524258-5a80d080-9c45-11eb-8a61-0755519a4f09.png)

And also enabled in the channel:
![image](https://user-images.githubusercontent.com/19649604/114524329-68365600-9c45-11eb-9d78-bd945e96d34b.png)

This is my created JAR file:
[ExampleTest.zip](https://github.com/nextgenhealthcare/connect/files/6302402/ExampleTest.zip)

This is what is inside the JAR File:
![image](https://user-images.githubusercontent.com/19649604/114524504-91ef7d00-9c45-11eb-8e83-77491e641f87.png)

I know that something is wrong with the process i've done since i was able to sucessfully import a JAR file from Talend.

**Comment by tonygermano** [comment]
I do not see anything obviously wrong with this. You may want to try setting up a real directory resource instead of using custom-lib and see if that makes a different. Custom-lib usage is generally considered deprecated.

**Reply by marvinschmidtke** [reply]
Hi thanks for your reply, unfortunatly i've tried it with a new resource aswell. I've also activated the new resource in the channel and got the same error.

Does it have something to do with my AdoptOpenJDK that i use for generating the the jar?

**Reply by brentmoen** [reply]
@marvinschmidtke I tried with the JAR you supplied and got the same error. Tried reproducing creating my own class, got the same error again. I saw your comment now and realized I had compiled with OpenJDK 15. When recompiling with Oracle Java 8, it works for me.

**Reply by tonygermano** [reply]
@bmoenng I think that comment belongs in its own answer! Can you still compile with OpenJDK 15, but set the target version for 8 (or 11 or whatever the mirth server is running on?)

**Comment by brentmoen** [comment, marked as answer]
Tried with the JAR you supplied and got the same error. Tried reproducing creating my own class and JAR, got the same error again. I saw your comment and realized I had compiled with OpenJDK 15. When building with Oracle Java 8, it works. I should note that I was running Connect with Oracle Java 8 when testing.

Per @tonygermano's question, I tested again building with OpenJDK 15, but setting the source and build target to Java 8. Tested with two JARs built with OpenJDK, one targeting 15 and one targeting 8 and confirmed that the former generates the error while the latter doesn't.

**Reply by brentmoen** [reply]
@marvinschmidtke Try compiling with `javac -d classes -source 8 -target 8 src/com/example/*`, let us know if that fixes it on your end
